File: src/main/java/com/praveen/biddingharbor/service/AuctionService.java
package com.praveen.biddingharbor.service;

import com.praveen.biddingharbor.dto.auction.*;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface AuctionService {

    AuctionResponse createAuction(
            String username,
            CreateAuctionRequest request);

    AuctionResponse updateAuction(
            Long auctionId,
            String username,
            UpdateAuctionRequest request);

    void deleteAuction(
            Long auctionId,
            String username);

    AuctionResponse getAuction(Long auctionId);

    List<AuctionResponse> getAllOpenAuctions();

    List<AuctionResponse> getMyAuctions(String username);

    AuctionResponse publishAuction(
            Long auctionId,
            String username);

    List<AuctionResponse> getPendingAuctions();

    AuctionResponse approveAuction(
            Long auctionId,
            String adminUsername);

    AuctionResponse rejectAuction(
            Long auctionId,
            String adminUsername,
            RejectAuctionRequest request);

    List<AuctionResponse> searchAuctions(
            SearchAuctionRequest request);

    Page<AuctionResponse> getAuctions(Pageable pageable);
}

File: bidding-harbor-frontend/src/services/auctionService.js
import axiosInstance from "../api/axiosInstance";

export const getAuctions = async () => {

    const response = await axiosInstance.get(
        "/auctions"
    );

    return response.data;
};

export const getAuction = async (id) => {

    const response = await axiosInstance.get(
        `/auctions/${id}`
    );

    return response.data;
};

export const getMyAuctions = async () => {

    const response = await axiosInstance.get(
        "/auctions/my"
    );

    return response.data;
};

export const createAuction = async (auctionData) => {

    const response = await axiosInstance.post(
        "/auctions",
        auctionData
    );

    return response.data;
};

export const updateAuction = async (id, auctionData) => {

    const response = await axiosInstance.put(
        `/auctions/${id}`,
        auctionData
    );

    return response.data;
};

export const publishAuction = async (id) => {

    const response = await axiosInstance.put(
        `/auctions/${id}/publish`
    );

    return response.data;
};

export const deleteAuction = async (id) => {

    const response = await axiosInstance.delete(
        `/auctions/${id}`
    );

    return response.data;
};

export const searchAuctions = async (searchParams) => {
    const response = await axiosInstance.get("/auctions/search", {
        params: searchParams
    });
    return response.data;
};